net/tsdial/tsdial.go

```diff
@@ -337,9 +338,10 @@ func (d *Dialer) UserDial(ctx context.Context, network, addr string) (net.Conn,
 		}
 		return d.NetstackDialTCP(ctx, ipp)
 	}
-	// TODO(bradfitz): netns, etc
-	var stdDialer net.Dialer
-	return stdDialer.DialContext(ctx, network, ipp.String())
+	if tsaddr.IsTailscaleIP(ipp.Addr()) {
+		return d.getPeerDialer().DialContext(ctx, network, ipp.String())
+	}
+	return d.SystemDial(ctx, network, ipp.String())
```

Comment:
I don't think systemdial is what you want, it implies dialing as though Tailscale isn't enabled. I guess that's what you're trying to do here in the default case but that will likely regress dialing out to subnets on macOS unless you extend the check above

Reply:
Reverted.